tests/integration/charmhub/test_charmhub_workflow.py

```diff
@@ -0,0 +1,83 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Full workflow tests for charms."""
+import time
+
+from craft_store.models import revisions_model
+
+from tests.integration.conftest import needs_charmhub_credentials
+
+
+@needs_charmhub_credentials()
+def test_full_charm_workflow(charm_client, charmhub_charm_name, fake_charms):
+    """A workflow test for uploading a full charm.
```

Comment:
What's a full charm?  A charm with resources?

Reply:
No, just "full" in the wrong space. It should be "A full workflow chest for uploading a charm." (Fixed in the latest commit)